### DISCUSSION TITLE
How do I use Pocketbase with Angular? I can't seem to import it in my code.

### DISCUSSION THREAD

**Discussion by SyedMSawaid** [pocketbase/pocketbase#758] [Q&A]
Has anyone tried Pocketbase with Angular? I am trying to import it but it doesn't show the option in intellisense.

**Comment by ganigeorgiev** [comment]
>  but it doesn't show the option in intellisense

What option? Do you mean that the JS SDK types are not loaded? 

If you can't see the types after importing the library, or you are not able to import it at all, then make sure that your TS config is properly setup and allow loading es modules - https://www.typescriptlang.org/docs/handbook/esm-node.html.

**Reply by ganigeorgiev** [reply]
@SyedMSawaid I've just tested it with the default angular starter template (aka. `ng new my-app`) and the JS SDK and its types are loaded just fine (in Sublime and Vim using their respective TS LSP plugins). This is how my `app.component.ts` looks like:

```js
import { Component } from '@angular/core';
import PocketBase from "pocketbase";

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'angular-app';

  ngOnInit(): void {
    // @todo you may want to move the client creation into a separate class/factory so that it can be injected as a service
    const pb = new PocketBase("http://127.0.0.1:8090");

    pb.records.getList("example", 1, 50).then((result) => {
      console.log(result);
    }).catch((err) => {
      console.log("Something went wrong:", err);
    });
  }
}
```

**Reply by SyedMSawaid** [reply]
![image](https://user-images.githubusercontent.com/34231494/195209024-0ba2dab6-0096-4cfc-ac66-9bc02543686d.png)

I get this error when I write `ng serve`

**Reply by ganigeorgiev** [reply]
I cannot reproduce it. Could you post here your base `tsconfig.json`?

**Reply by ganigeorgiev** [reply]
@SyedMSawaid Additionally, could you also try running `ng version`? This is mine:

```
Angular CLI: 14.2.5
Node: 16.16.0
Package Manager: npm 8.11.0 
OS: linux x64

Angular: 14.2.5
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1402.5
@angular-devkit/build-angular   14.2.5
@angular-devkit/core            14.2.5
@angular-devkit/schematics      14.2.5
@schematics/angular             14.2.5
rxjs                            7.5.7
typescript                      4.7.4
```

**Reply by ganigeorgiev** [reply]
My guess is that your `tsconfig.json` is generated with an older `ng` version or app template. As mentioned previously, make sure that your `tsconfig.json` contains up-to-date `moduleResolution` and `node` properties. The default generated from the starter template (and that works for me) are:

```
...
"moduleResolution": "node",
"importHelpers": true,
"target": "es2020",
"module": "es2020",
"lib": [
  "es2020",
  "dom"
]
...
```
(_more on the available module resolution strategies you could find in the official ts documentaiton - https://www.typescriptlang.org/docs/handbook/module-resolution.html#module-resolution-strategies_)

If with the above still doesn't work, make sure to upgrade to the latest ng version (14.2.5) and try creating a new blank app with `ng new my-app`.

**Reply by SyedMSawaid** [reply]
```json
/* To learn more about this file see: https://angular.io/config/tsconfig. */
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es2017",
    "module": "es2020",
    "lib": [
      "es2020",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "enableI18nLegacyMessageIdFormat": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
}
```
TSConfig ⬆


`ng version` ⬇
```bash
Angular CLI: 13.3.9
Node: 14.18.1
Package Manager: npm 6.14.15
OS: win32 x64

Angular: 13.3.11
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1303.9
@angular-devkit/build-angular   13.3.9
@angular-devkit/core            13.3.9
@angular-devkit/schematics      13.3.9
@angular/cli                    13.3.9
@schematics/angular             13.3.9
rxjs                            7.5.7
typescript                      4.6.4
```

**Reply by SyedMSawaid** [reply]
I am trying it by updating my angular CLI. Hope it works.

**Reply by ganigeorgiev** [reply]
@SyedMSawaid I've tried to emulate your setup (including the old Node 14) but I wasn't able to reproduce the error. In any case, I'd recommend upgrading:
- node to 16+ (node 14 is no longer actively maintained - https://endoflife.date/nodejs)
- angular/cli to 14+
- your project dependencies (including typescript), eg. `npm update`

Here are the steps that I've tried and works for me:

1. Download and install the latest angular/cli - `npm install @angular/cli -g`

2. Create a new application -  `ng new my-app`

3. Install the PocketBase JS SDK - `npm install pocketbase --save`

4. Import the sdk into  `app.component.ts`:
```js
import { Component } from '@angular/core';
import PocketBase from "pocketbase";

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'angular-app';

  ngOnInit(): void {
    // @todo you may want to move the client creation into a separate class/factory so that it can be injected as a service
    const pb = new PocketBase("http://127.0.0.1:8090");

    pb.records.getList("example", 1, 50).then((result) => {
      console.log(result);
    }).catch((err) => {
      console.log("Something went wrong:", err);
    });
  }
}
```

5. Run `npm run start` (or `ng serve`).

---

If after the above you are still getting the same error, please provide a minimal reproducible git repo and I'll have a look at it.
